sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/recovery/monitor/TimedFailureMonitor.java

```diff
@@ -69,7 +78,12 @@ public boolean hasFailed(TaskInfo terminatedTask) {
                 + taskExpiredTime + " which is " + now.after(taskExpiredTime));
 
         if (now.after(taskExpiredTime)) {
-            FailureUtils.markFailed(terminatedTask);
+            try {
+                PodInstance podInstance = TaskUtils.getPodInstance(configStore, terminatedTask);
+                FailureUtils.markFailed(podInstance, stateStore);
+            } catch (TaskException e) {
+                log.error("Failed to get pod instance", e);
```

Comment:
- could say "Failed to get pod instance to mark as failed" to give more context in logs
- could log the relevant task in this error (unless it's already logged in the exception...)